### ISSUE TITLE
QuickJS Support

### ISSUE THREAD

**Issue by akrymski** [ultralight-ux/Ultralight#304]
Would be great to support alternatives to JavaScriptCore, such as the lightweight [QuickJS](https://bellard.org/quickjs/).

**Comment by adamjs** [comment]
Yeah I agree, or even disable JS completely by stubbing out the JSC interface.

FWIW, JavaScriptCore is pretty tightly coupled to WebCore at the moment (if you want to poke around the source, the main interface between WebCore and JSC is [https://github.com/ultralight-ux/WebCore/blob/master/Source/WebCore/bindings/js/ScriptController.cpp](https://github.com/ultralight-ux/WebCore/blob/master/Source/WebCore/bindings/js/ScriptController.cpp)). 

An easier target for now is to try to selectively reduce the size of JavaScriptCore-- we can compile it without support for the various JIT compilers and WebAssembly which shaves about 15MB off the binary. I've gone ahead and done this in the latest release for 1.2.0 stable (see the `min` packages).

**Comment by akrymski** [comment]
Thanks, I'll check it out. If I wanted to implement an alternative language
to JS would that also be the place to do it? In particular I'm interested
in being able to manipulate the DOM directly in C instead of using a
scripting language. That would open up possibilities to write native apps
in new languages like Rust and Nim using WebCore for rendering without the
overhead of JS.

If there is a way to selectively eliminate WebCore features to support some
new-ish subset of HTML (5 only) and CSS (eg flexbox and maybe grid should
be all you need for layout for example, drop support for floats etc) would
be interesting.

Are you using the embedded WebKit branch btw? Seems relevant.
